src/heitech.zer0mqXt.core/Infrastructure/SocketConfiguration.cs

```diff
@@ -16,47 +16,50 @@ internal abstract class SocketConfiguration : IEquatable<SocketConfiguration>
         ///<summary>
         /// Uses the BasicLogger implementation by default which only Logs to the console
         ///</summary>
-        public ILogger Logger { get; set; }
-
+        public ILogger Logger 
+        {
+            get => _logger ?? new BasicLogger();
```

Comment:
It seems a bit odd to me that, as long as the backing field is `null`, we return a new instance of `BasicLogger` each time we invoke the `get` accessor of the property.
Did you mean to also assign the new instance when the backing field is `null` via `??=`?

Reply:
yes indeed, that is a mistake :D.

Reply:
done